docs/examples/grpc/README.md

```diff
@@ -1,93 +1,141 @@
-
-For your own application you may or may not want to do this.  If you prefer to
-forward encrypted traffic to your POD and terminate TLS at the gRPC server
-itself, add the ingress annotation `nginx.ingress.kubernetes.io/backend-protocol: "GRPCS"`.
-
-### Step 2: the kubernetes `Service`
-
-```sh
-$ kubectl create -f svc.yaml
-```
-
-Here we have a typical service. Nothing special, just routing traffic to the
-backend application on port `50051`.
-
-### Step 3: the kubernetes `Ingress`
-
-```sh
-$ kubectl create -f ingress.yaml
-```
-
-A few things to note:
-
-1. We've tagged the ingress with the annotation
-   `nginx.ingress.kubernetes.io/backend-protocol: "GRPC"`.  This is the magic
-   ingredient that sets up the appropriate nginx configuration to route http/2
-   traffic to our service.
-1. We're terminating TLS at the ingress and have configured an SSL certificate
-   `fortune-teller.stack.build`.  The ingress matches traffic arriving as
-   `https://fortune-teller.stack.build:443` and routes unencrypted messages to
-   our kubernetes service.
+2. You have a domain name such as `example.com` that is configured to route traffic to the ingress controller.
+3. You have the nginx-ingress controller installed as per docs for gRPC support.
+4. You have a backend application running a gRPC server and listening for TCP traffic.  If you want, you can use <https://github.com/grpc/grpc-go/blob/91e0aeb192456225adf27966d04ada4cf8599915/examples/features/reflection/server/main.go> as an example.
+5. You're also responsible for provisioning an SSL certificate for the ingress. So you need to have a valid SSL certificate, deployed as a Kubernetes secret of type tls, in the same namespace as the gRPC application.
+
+### Step 1: Create Kubernetes `Deployment` for gRPC app
+
+- Make sure your gRPC application pod is running and listening for connections. For example you can try a kubectl command like this below :
+  ```
+  $ kubectl get po -A -o wide | grep go-grpc-greeter-server
+  ```
+- If you have a gRPC app deployed in your cluster, then skip further notes in this Step 1, and continue from Step 2 below. 
+
+- As an example grpc application, we can use this official gRPC app <https://github.com/grpc/grpc-go/blob/91e0aeb192456225adf27966d04ada4cf8599915/examples/features/reflection/server/main.go> .
+
+- To create a container image for this app, you can use [this Dockerfile](../../../images/go-grpc-greeter-server/rootfs/Dockerfile). 
+
+- Currently, an image is built using this Dockerfile and published on hub.docker.com as `longwuyuan/go-grpc-greeter-server`. Soon, this will change, and then this document will be updated accordingly.
+
+- If you use the Dockerfile mentioned above, to create a image, then given below is an example of a Kubernetes manifest, to create a deployment resource, that uses that image. If needed, then edit this manifest to suit your needs. Assuming the name of this yaml file is  `deployment.go-grpc-greeter-server.yaml` ;
```

Comment:
I didn't get the need of the last sentence (Assuming the name of...)

As we are not referencing anything the file name deployment.go-grpc-greeter-server.yaml anywhere else, we can:
* Show the kubectl apply -f this file name OR
* do somethng like (and personally I prefer this one, as someone can just copy/paste it

```
cat <<EOF | kubectl apply -f -
apiVersion: apps/v1
   kind: Deployment
   metadata:
     labels:
       app: go-grpc-greeter-server
     name: go-grpc-greeter-server
   spec:
     replicas: 1
     selector:
       matchLabels:
         app: go-grpc-greeter-server
     template:
       metadata:
         labels:
           app: go-grpc-greeter-server
       spec:
         containers:
        - image: longwuyuan/go-grpc-greeter-server
           name: go-grpc-greeter-server
           ports:
           - containerPort: 50051
EOF
```

Reply:
Changed to cat <<EOF